src/main/java/jenkins/plugins/office365connector/FactsBuilder.java

```diff
@@ -82,11 +122,19 @@ public void addCompletionTime() {
         facts.add(new Fact(NAME_COMPLETION_TIME, TimeUtils.dateToString(countCompletionTime())));
     }
 
-    private long countCompletionTime() {
-        long duration = run.getDuration() == 0L
-                ? System.currentTimeMillis() - run.getStartTimeInMillis()
-                : run.getDuration();
-        return run.getStartTimeInMillis() + duration;
+    private long getRunDuration()
```

Comment:
should be defined below method which invokes it

Reply:
OK